CONTRIBUTING updates for CUDA (#159)
nv-morpheus/SRF: CONTRIBUTING.md

directory:

1. Generate a code coverage report and ensure your additions are covered.

>    1. take note of the CUDAToolkit setup in the Build section below

Reviewer: ```suggestion
   1. take note of the CUDA Toolkit setup in the Build section below
```

>    1. take note of the CUDA Toolkit setup in the Build section below

Reviewer: ```suggestion
   1. Take note of the CUDA Toolkit setup in the Build section below
```

### Build in a Conda Environment

#### CUDA Toolkit Setup

SRF uses the CUDA Toolkit. If you have multiple versions installed on your host, then some care needs to be taken in your environment.
For example, you may see the following error when activating the srf conda environment:

`Version of installed CUDA didn't match package`

Some options:

> - Set the variable CUDA_HOME to the desired CUDA install

Reviewer: ```suggestion
- Set the variable `CUDA_HOME` to the desired CUDA install
```

>   - This option is good for overriding the value set in PATH if you have multiple installs

Reviewer: ```suggestion
  - This option is good for overriding the value set in `PATH` if you have multiple installs
```

>   - This will also get rid of the warning messages when activating conda
> 

Reviewer: ```suggestion
  - This will also get rid of the warning messages when activating conda
  - Note: This must be set before calling `conda activate` and will only work for the lifetime of that shell session. For that reason, it's best to configure this in your `.bashrc` or similar configuration file.

```

> - Set the CMake CUDA variable -DCUDAToolkit_ROOT

Reviewer: ```suggestion
- Set the CMake CUDA variable `-DCUDAToolkit_ROOT`
```

>   - For example, you can set `-DCUDAToolkit_ROOT=/usr/local/cuda-11.5` to tell CMake to use that install

Reviewer: ```suggestion
  - For example, you can set `-DCUDAToolkit_ROOT=/usr/local/cuda-11.5` to tell CMake to use your CUDA 11.5 install
```

>   - This will work even if the nvcc_linux-64 conda package is uninstalled

Reviewer: ```suggestion
  - This will work even if the `nvcc_linux-64` conda package is uninstalled
```
